lib/watir/elements/select.rb

```diff
@@ -161,6 +196,23 @@ def select_by(how, str_or_rx)
       select_matching(found)
     end
 
+    def select_by!(how, str_or_rx, number)
+      script = how == :text ? text_script : value_script
+
+      if str_or_rx.is_a? Regexp
+        js_rx = str_or_rx.inspect.sub('\\A','^').sub('\\Z','$').sub('\\z','$').sub(/^\//,'').sub(/\/[a-z]*$/,'').gsub(/\(\?#.+\)/, '').gsub(/\(\?-\w+:/,'(')
```

Comment:
Don't get me wrong, but regexp engines in JavaScript and Ruby are much more different. For example, there is no support for negative lookbehind, POSIX expressions, etc. Also, there are different flags.

With that said, I'd rather not support it at all than saying "you could try, but there is no way we can guarantee it will work" :)